cinema/views.py

```diff
@@ -24,21 +29,29 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly, IsAuthenticated)
```

Comment:
The `permission_classes` for `GenreViewSet` includes both `IsAdminOrIfAuthenticatedReadOnly` and `IsAuthenticated`. This is redundant because `IsAdminOrIfAuthenticatedReadOnly` already handles both admin and authenticated user permissions. You can remove `IsAuthenticated` from this tuple.